### PR TITLE
Add pagination when listing users of a team

### LINKED ISSUE #11: List of team members do not support pagination
Here:
https://github.com/camptocamp/github_pki/blob/860f9f335e17cb8c5d4ba874816fb05fe0f86b81/main.go#L133

There is a single call without handling pagination. When used on a team with more than 30 members (default github's pagination), only 30 members will have their key in the file.

### PR DESCRIPTION
**Pull request by guewen**
Allows to have more than 30 users in a team.

Fixes #11

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -130,8 +130,21 @@ func (p *gitHubPki) getTeamUsers() (err error) {
...
+				var ghUsers []github.User
+				opt := &github.TeamListTeamMembersOptions{
+					ListOptions: github.ListOptions{PerPage: 100},
+				}
+				for {
+					pageUsers, resp, err := p.Client.Teams.ListTeamMembers(context.Background(), *team.ID, opt)
+					checkErr(err, "Failed to list team members for team "+*team.Name+": %v")
+					for _, u := range pageUsers {
+						ghUsers = append(ghUsers, *u)
+					}
```
Comment (raphink) [inline main.go]: iirc `ghUsers = append(ghUsers, pageUsers...)` would work

File: main.go [hunk 2]
```diff
@@ -130,8 +130,21 @@ func (p *gitHubPki) getTeamUsers() (err error) {
...
+					if resp.NextPage == 0 {
+						break
+					}
+					opt.Page = resp.NextPage
```
Comment (raphink) [inline main.go]: iirc you could write

```go
if opt.Page = resp.NextPage; opt.Page == 0 {
  break
}
```

**Comment by guewen** [comment]
@raphink Thanks! I confirm both your changes work, added a fixup commit

File: main.go [hunk 3]
```diff
@@ -130,8 +130,18 @@ func (p *gitHubPki) getTeamUsers() (err error) {
 	for _, team := range teams {
 		for _, t := range p.Config.Teams {
 			if *team.Name == t {
-				ghUsers, _, err := p.Client.Teams.ListTeamMembers(context.Background(), *team.ID, nil)
-				checkErr(err, "Failed to list team members for team "+*team.Name+": %v")
+				var ghUsers []*github.User
+				opt := &github.TeamListTeamMembersOptions{
+					ListOptions: github.ListOptions{PerPage: 100},
```
Comment (raphink) [inline main.go]: If we're following pagination, do we really need to increase the amount of results per page?

File: main.go [hunk 4]
```diff
@@ -130,8 +130,18 @@ func (p *gitHubPki) getTeamUsers() (err error) {
...
 		for _, t := range p.Config.Teams {
 			if *team.Name == t {
-				ghUsers, _, err := p.Client.Teams.ListTeamMembers(context.Background(), *team.ID, nil)
-				checkErr(err, "Failed to list team members for team "+*team.Name+": %v")
+				var ghUsers []*github.User
+				opt := &github.TeamListTeamMembersOptions{
+					ListOptions: github.ListOptions{},
+				}
+				for {
+					pageUsers, resp, err := p.Client.Teams.ListTeamMembers(context.Background(), *team.ID, opt)
```
Comment (raphink) [inline main.go:134]: I think `opt` could remain `nil` as it was before since we're not overriding anything.

File: main.go (hunk 3 above)
Comment (guewen) [reply main.go]: I used the same value as the listing of teams done above, but you are right, it's probably better to keep the default value. BTW, do you want me to update the listing of teams with the same changes as I did there to be consistent?

Comment (raphink) [reply main.go]: Sure, good idea

File: main.go (hunk 4 above)
Comment (guewen) [reply main.go:134]: We have to modify the `Page` of the `opt` object for the calls after the first page.

File: main.go (hunk 3 above)
Comment (guewen) [reply main.go]: Done

**Pull request merged**